roadmap/README.md

```diff
@@ -0,0 +1,148 @@
+  from the rich high quality catalog of reusable components:
+
+  * Quickly build and interact with sophisticated `Pipelines`
+  * Be able to port `Pipelines` to any Tekton conformant system
+  * Be able to use multiple Tekton conformant systems instead of being locked into one
+    or being forced to build glue between multiple completely different systems
+  * Use an ecosystem of tools that know how to interact with Tekton components, e.g.
+    IDE integrations, linting, CLIs, security and policy systems
+
+## Roadmap
+
+### Beta and GA
+
+In 2019 we got to the point where we had several projects built on top of Tekton
+that were craving additional API stability, so we started a push toward having
+a beta release of Tekton. We have
+[created a plan](https://docs.google.com/document/d/1H8I2Rk4kLdQaR4mV0A71Qbk-1FxXFrmvisEAjLKT6H0/edit)
+which defines what Beta means for Tekton Pipelines, and
+[through our beta working group](https://github.com/tektoncd/community/blob/master/working-groups.md#beta-release)
+we are working towared a beta release in early 2020.
+
+The initial beta release
+[does not include all Tekton Pipeline components](https://docs.google.com/document/d/1H8I2Rk4kLdQaR4mV0A71Qbk-1FxXFrmvisEAjLKT6H0/edit#heading=h.t0sc4hdrr5yq). After our initial
+beta release, we will then work toward:
+
+1. Beta for the remaining components
+1. Beta for [Tekton Triggers](#triggers)
+1. GA for both Tekton Pipelines and Tekton Triggers
+
+As the project matures, we also require:
+
+1. A website that provides a good landing page for users (tekton.dev)
+1. Solid, high quality onboarding and documentation
+
+### "Feature Complete" Tekton
+
+Today Tekton can do basically anything you want it to, but there are a few features
+still missing or not quite where we want them. Once we have these features, folks
+should be able to use Tekton for all of their CI/CD use cases.
+
+Deciding what to support and what to expect users to write into their own `Tasks`
+is a constant balancing act, and we want to be careful and deliberate with what we
+include.
+
+Features we don't have or aren't yet 100% satisfied with for "feature complete"
+Tekton:
+
+- [Failure strategies](https://github.com/tektoncd/pipeline/issues/1684)
+- [Conditional execution](https://github.com/tektoncd/pipeline/blob/master/docs/conditions.md)
+  - we will continue to iterate on this functionality as we add failure strategies
+- [Pause and resume](https://github.com/tektoncd/pipeline/issues/233)
+- [Partial execution](https://github.com/tektoncd/pipeline/issues/50)
+- [Notifications](https://github.com/tektoncd/pipeline/issues/1740)
+- [`Task` Versioning](https://github.com/tektoncd/pipeline/issues/1839) - Also required
+  for [Catalog](#catalog) success
+- [Alternative Task Implementations](https://github.com/tektoncd/pipeline/issues/215)
+- [Local Execution](https://github.com/tektoncd/pipeline/issues/235)
+- Testing and debugging frameworks
```

Comment:
:thinking: ?

Reply:
ah! maybe I need to write up my thoughts here a bit more. I really like how projects like OPA come with their own built in support for testing (https://www.openpolicyagent.org/docs/latest/policy-testing/) and as @ImJasonH and I hinted at super briefly in that russian doll talk, we could potentially use the entrypoint binary to expose some cool debugging support (e.g. letting users step through steps)

Reply:
oh yeah also https://github.com/tektoncd/pipeline/issues/1289

Reply:
now we've also got https://github.com/tektoncd/pipeline/issues/2069

Reply:
@bobcatfish 

@nikhil-thomas and I would be working on it. :smile_cat: